### PR TITLE
[NUI][CheckBox] modification of the guide content

### PR DESCRIPTION
**Pull request by mgrudzinska**
modification of the checkbox section in the NUI guide

### CODE REVIEW

File: docs/application/dotnet/guides/nui/nui-components/CheckBox.md [hunk 1]
```diff
@@ -1,77 +1,205 @@
 # CheckBox
-CheckBox is a common component and describes what action will occur when you select it.  
-A CheckBox can only contain an icon, and can be created using style.
 
-![CheckBox1](./media/CheckBox1.png)
+Checkbox is a UI component connected with the click events. The base class for a `CheckBox` class is a [Tizen.NUI.Components.Button](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.Components.Button.html), which properties can be used to specify the checkbox. In opposite to a button, that can contain both an icon and a text, a checkbox can only contain an icon. The default checkbox created with NUI is as follows:
```
Comment (annie-samsung) [inline docs/application/dotnet/guides/nui/nui-components/CheckBox.md:3]: Checkbox is a UI component connected with the click events. The base class for a `CheckBox` class is a Tizen.NUI.Components.Button, you can use the base class properties to specify the checkbox. Unlike a button that can contain both an icon and a text, a checkbox can only contain an icon. The default checkbox created with NUI is as follows:


@mijin85cho I have removed the URL to be consistent with other pages, thoughts?
@mgrudzinska , which properties can be used to specify the checkbox. ? whose properties base class? Please check my rephrase on top?

Comment (mijin85cho) [reply docs/application/dotnet/guides/nui/nui-components/CheckBox.md:3]: @annie-samsung 

The base class for a CheckBox class is a Tizen.NUI.Components.Button, you can use the base class properties to specify the checkbox.

I think the way two sentences are connected is weird. Is it OK to just add two sentences without conjunction?

The link to the related API helps user to access to more information, so you can keep it. If you are a user, the link would be helpful to you..

File: docs/application/dotnet/guides/nui/nui-components/CheckBox.md [hunk 2]
```diff
@@ -1,77 +1,205 @@
...
+To disable the checkbox completely use the following code:
+   ```cs
+   // CheckBox is disabled
+   _checkBox.IsEnabled = false;
+   ```
+This sets the checkbox state to `Disabled` or `DisabledSelected` depending on the `IsSelected` value. The preceding options change the checkbox appearance as follows:
+
+| ![CheckBox_dis](./media/CheckBox_dis.gif)                   | ![CheckBox_dis](./media/CheckBox_yellow.gif)             | ![CheckBox_dis](./media/CheckBox_green.gif)               |
+|:-----------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------------------:|
+| ```IsSelectable``` set to `false`<br>`IsSelected` set to `true` | `IsEnabled` set to `false`<br>`IsSelected` set to `true` | `IsEnabled` set to `false`<br>`IsSelected` set to `false` |
```
Comment (mijin85cho) [inline docs/application/dotnet/guides/nui/nui-components/CheckBox.md:69]: In this page, I'm a little bit confused below point. 

1. In my understanding, in the first case the button can be clicked, but button value is not changeable (`IsSelectable` = `false`), and the button value is "checked" (`IsSelected` = `true`).
If so, isn't the button always in the checked state even how many times it is clicked?
In the image, first it is in checked state, and changed to unchecked state at the click. After that it keeps unchecked state even it is clicked again many times. So I'm a little bit confused. If I understood something wrong, would you please let me know?

2. In the second and the third case, the buttons are disabled, 
so even it is clicked the red image doesn't appear, means not clickable.
The second button is checked, so it is in yellow. (DisabledSelected = _URL + "yellow.png").
The third button is not checked, so it is in green. (Disabled         = _URL + "green.png",)

   However, in the second image, the yellow seems overlayed on blue, so it looks like yellowish green. Is it possible to strengthen the yellow in the image?

Comment (mijin85cho) [inline docs/application/dotnet/guides/nui/nui-components/CheckBox.md:69]: @annie-samsung I think you missed this.

```suggestion
| `IsSelectable` set to `false`<br>`IsSelected` set to `true` | `IsEnabled` set to `false`<br>`IsSelected` set to `true` | `IsEnabled` set to `false`<br>`IsSelected` set to `false` |
```

File: docs/application/dotnet/guides/nui/nui-components/CheckBox.md [hunk 3]
```diff
@@ -1,77 +1,205 @@
...
-    // Do something in response to state change
-}
-```
+
+When you click a checkbox, the `CheckBox` instance receives a clicked event. You can declare the clicked event handler as follows:
+   ```cs
+   CheckBox _checkBox = new CheckBox();
+   _checkBox.Clicked += OnClicked;
+   ```
+where `OnClicked` a function defined by the user:
```
Comment (annie-samsung) [inline docs/application/dotnet/guides/nui/nui-components/CheckBox.md:196]: where `OnClicked` is a function defined by the user:

File: docs/application/dotnet/guides/nui/nui-components/CheckBox.md (hunk 2 above)
Comment (annie-samsung) [reply docs/application/dotnet/guides/nui/nui-components/CheckBox.md:69]: @mijin85cho, I am still reviewing, looks like I accidently submitted single review or so. Thanks for pointing it out.

File: docs/application/dotnet/guides/nui/nui-components/CheckBox.md (hunk 1 above)
Comment (mijin85cho) [reply docs/application/dotnet/guides/nui/nui-components/CheckBox.md:3]: > @mgrudzinska , which properties can be used to specify the checkbox. ? whose properties base class? Please check my rephrase on top?

I think the "which" means "Tizen.NUI.Components.Button", in the previous sentence.

As you can see in https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.Components.CheckBox.html,
the CheckBox inherited https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.Components.Button.html, Class Button.
So the CheckBox inherited properties of the Button also, which means CheckBox can _use_ the the properties such as 'IsEnabled', or 'Clicked' event from Button.
(If you see https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.Components.CheckBox.html, you'd not see `IsEnabled` or `Clicked` at there, but they are used in the following code. )

@mgrudzinska If I misunderstood something wrong, please fix it.

File: docs/application/dotnet/guides/nui/nui-components/CheckBox.md (hunk 2 above)
Comment (mijin85cho) [reply docs/application/dotnet/guides/nui/nui-components/CheckBox.md:69]: @annie-samsung I see. Thanks for the reply.

**Review by annie-samsung — COMMENTED** [review]
No further comments

**Comment by annie-samsung** [comment]
@mgrudzinska 
Gentle Reminder, could you please check the comments.

File: docs/application/dotnet/guides/nui/nui-components/CheckBox.md (hunk 1 above)
Comment (mgrudzinska) [reply docs/application/dotnet/guides/nui/nui-components/CheckBox.md:3]: @mijin85cho yes, you are right.
The properties of the checkbox class are inherited from the button class. 

@annie-samsung, I've changed the sentence as follows:
The base class for a `CheckBox` class is a [Tizen.NUI.Components.Button](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.Components.Button.html). The `Button` properties can be used to specify the checkbox.

The button properties should be specified in the button section, so I don't want to repeat this here. 
The URL seems to be consistent with other pages of NUI guide.

File: docs/application/dotnet/guides/nui/nui-components/CheckBox.md (hunk 2 above)
Comment (mgrudzinska) [reply docs/application/dotnet/guides/nui/nui-components/CheckBox.md]: please check now.
1. After reconsideration, I don't think it's a good idea to use IsSelectable=false and IsSelected=true. the behaviour is not obvious (it depends on the order of commands). I changed IsSelected to false.  Please let me know whether it is clear now.
2. I've changed the opacity. is there anything else that requires clarification?

**Comment by mgrudzinska** [comment]
thank you for comments. I answered some of them and incorporated the rest.

File: docs/application/dotnet/guides/nui/nui-components/CheckBox.md (hunk 2 above)
Comment (mijin85cho) [reply docs/application/dotnet/guides/nui/nui-components/CheckBox.md]: Thanks for the update. I read the updated parts and now they are clear to me. I hope @annie-samsung also can check the updated parts and let us know the opinion.

**Review by annie-samsung — APPROVED** [review]
looks good, hence approving

**Comment by annie-samsung** [comment]
Hello Mira (@mgrudzinska)

I approved the PR, but the had one feedback. The images are not of uniform sizes. Especially, Line 135 is very big.

Line 53, 62 images have lot of white space around.

**Comment by mgrudzinska** [comment]
Hi @annie-samsung 
Thank you for all your comments
I unified the gifs sizes. 
The last one is a bit bigger, since it is declared bigger in the code.

**Review by annie-samsung — COMMENTED** [review]
Thank you Mira, looks good, approving again.

**Pull request merged**